data-processing/entities/terms_v1/extractor.py

```diff
@@ -0,0 +1,123 @@
+                    if term == word:
+                        ###Storing as a list because we want to keep the original text_segment_string for debugging (context_tex)
+						matches_tuples.append([term, text_segment_string])
+            for word2 in text_segment_string_not_split_on_dashes:
+                for term in terms:
+                    if term == word2:
+                        matches_tuples.append([term, text_segment_string])
+            for term in terms:
+                term_list = re.split("\s+", term)
+                if len(term_list) > 1:
+                    if term in text_segment_string:
+                        matches_tuples.append([term, text_segment_string])
+            # matches = list(matches)
+            # matches.add(matches_tuples)
+
+            for i in matches_tuples:
+                matches.add(tuple(i))
+
+            ###Getting the start and end indices of the matched terms
+            for match in matches:
+                start_index_of_string = text_segment_string.find(match[0])
+                end_index_of_string = start_index_of_string + len(match[0]) - 1
+                matched_terms.append(match[0])
+                matched_terms_context_str.append(match[1])
+                matched_terms_start.append(text_segment.tex_start + start_index_of_string)
+                matched_terms_end.append(text_segment.tex_start + end_index_of_string)
+
+        data_frame_dict = {"term": matched_terms, "start": matched_terms_start, "end": matched_terms_end, "context_str": matched_terms_context_str}
+        data_frame = pd.DataFrame(data_frame_dict).sort_values(by=['end'])
+        ###Getting rid of cases where both "Neural Network" and "Neural Networks"
+        ###are included (when they start at the same index, i.e. same word except with an extra "s")
+        data_frame = data_frame.drop_duplicates(subset=['start'], keep='last')
+        data_dict = data_frame.to_dict('records')
+        sorted_string = json.dumps(data_dict, indent=4, sort_keys=False)
+        json_data = json.loads(sorted_string)
+
+        index = 0
+        for term_object in json_data:
+        # Everything after here, you replace with your extraction of terms
+		# from plaintext segments.t - DEFAULT_CONTEXT_SIZE : end + DEFAULT_CONTEXT_SIZE]
+            yield Term(
+			text=term_object["term"],
+			start=term_object["start"],  # start position of the term in the TeX
+			end=term_object["end"],  # end position
+			id_=str(index),  # unique ID for each term (index)
+			tex_path=tex_path,  # tex_path (what was passed into the function)
+			tex=tex,  # includes all of the spaces and the LaTeX junk
+			##context_tex=term_object["context_str"],  # tex + 20 characters before + 20 characters after
+            # Assuming DEFAULT_CONTEXT_SIZE is 20 characters
+                        context_tex = tex[start - DEFAULT_CONTEXT_SIZE : end + DEFAULT_CONTEXT_SIZE]
+		    )
+            index += 1
+
+
+
+
+
+
+
+
```

Comment:
nit pick: remove unnecessary whitespace

Reply:
@jusheen are you using Visual Studio Code? If not, go ahead and download it, enable formatting on saving, and then save the code. That should take care of these dangling newlines, and fix up indents automatically 🎆